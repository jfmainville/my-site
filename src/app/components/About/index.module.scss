@import "../../variables.module.scss";

.About {
  display: flex;
  flex-direction: column;
  background-color: $secondary-color;
}

.AboutSection {
  display: flex;
  align-items: center;
  flex-direction: row;
}

@media (max-width: $small-screen) {
  .AboutSection {
    flex-direction: column;
  }
}

.AboutSectionText {
  display: flex;
  text-align: end;
  flex-direction: row;
  padding: 0 0 0 2rem;
}

.TextHeading {
  margin: 2rem;
}
