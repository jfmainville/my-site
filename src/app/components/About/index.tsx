import styles from "./index.module.scss";
import Image from "next/image";

const About = () => {
  return (
    <section id={"about"} className={styles.About}>
      <h1 className={styles.TextHeading}>About</h1>
      <div className={styles.AboutSection}>
        <p className={styles.AboutSectionText}>
          I am an experienced professional with expertise in cloud security,
          DevOps, and system administration. My career spans roles as a Cloud
          Security Specialist, Technical Product Owner, DevOps Specialist,
          Security Analyst, and System Administrator across diverse
          environments. I excel in designing and implementing secure IT
          solutions, automating processes, and optimizing performance to
          strengthen operational resilience and meet business objectives.
        </p>
        <Image src="/images/profile.png" width={500} height={500} alt="" />
      </div>
    </section>
  );
};

export default About;
