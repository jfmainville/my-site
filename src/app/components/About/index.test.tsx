import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import About from "./index";

describe("<About/> Test Suite", () => {
  it("should render the <About/> component unchanged", () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });

  it("should show my profile description in the <About /> component", () => {
    render(<About />);
    const profileDescription = screen.getByText(
      "I am an experienced professional",
      {
        exact: false,
      },
    );
    expect(profileDescription).toBeInTheDocument();
  });

  it("should show my profile picture in the <About /> component", () => {
    render(<About />);
    const profilePicture = screen.getByRole("presentation");
    expect(profilePicture).toBeInTheDocument();
  });
});
