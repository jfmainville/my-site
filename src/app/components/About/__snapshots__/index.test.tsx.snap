// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About/> Test Suite should render the <About/> component unchanged 1`] = `
<div>
  <section
    class="About"
    id="about"
  >
    <h1
      class="TextHeading"
    >
      About
    </h1>
    <div
      class="AboutSection"
    >
      <p
        class="AboutSectionText"
      >
        I am an experienced professional with expertise in cloud security, DevOps, and system administration. My career spans roles as a Cloud Security Specialist, Technical Product Owner, DevOps Specialist, Security Analyst, and System Administrator across diverse environments. I excel in designing and implementing secure IT solutions, automating processes, and optimizing performance to strengthen operational resilience and meet business objectives.
      </p>
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fprofile.png&w=1080&q=75"
        srcset="/_next/image?url=%2Fimages%2Fprofile.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fprofile.png&w=1080&q=75 2x"
        style="color: transparent;"
        width="500"
      />
    </div>
  </section>
</div>
`;
